### ISSUE TITLE
How to initiate (put) index template with `DocType` mappings?

### ISSUE THREAD

**Issue by vaniakosmos** [elastic/elasticsearch-dsl-py#927]
How to create index template with `DocType`?

Currently I use "raw" API to create index template:
```python
es.indices.put_template('template_1', {
    "template": "words_*",
    "order": 0,
    "settings": {
        "number_of_shards": 1
    },
    "mappings": {
        "doc": {
            "properties": {
                "text": {
                    "type": "text",
                    "analyzer": "some_language"
                }
            }
        }
    }
})
```

But I don't want to retype each field from my `DocType`...
I've found [IndexTemplate](https://elasticsearch-dsl.readthedocs.io/en/latest/persistence.html#indextemplate) in docs and source code, but it applied to some strange `Document` instead of `DocType` (and used by calling protected field for some reason `Doc._index.as_template()`).
Is `IndexTemplate` part of new API (accidentally committed to 6.1.0 version)? When this API will be available?

**Comment by HonzaKral** [comment]
I am about to release a new version which comes with examples (0) to do what you need, it will be available as soon as `6.2.0` comes out, please follow #912 for notifications.

For now you can always instantiate `IndexTemplate` and then call `my_index_template.doc_type(MyDocType)` to include the `DocType` mappings in it automatically.

Hope this helps!

0 - https://github.com/elastic/elasticsearch-dsl-py/blob/master/examples/alias_migration.py#L57-L71

**Comment by vaniakosmos** [comment]
Superb! This is exactly what I looked for all day
```
index_template = IndexTemplate('template_1', 'index_*')
index_template.doc_type(Doc)
index_template.settings(number_of_shards=1)
index_template.save()
```
